core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -315,6 +315,12 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val ENABLE_EXECUTOR_TREE_AGGREGATE = ConfigBuilder("spark.executor.treeAggregate")
```

Comment:
I don't think I'd add another config for this - it becomes yet another of 100 things that 99% of users won't know about. I'd imagine there is a generally better or worse behavior we can agree on

Reply:
For simpler treeAggregate operations, this might add an additional overhead (not sure if it is worth it or not) if the default behavior is to spawn a spark job for the last fold operation.

Reply:
I have a similar feeling w/ Sean's comment here. Should we maybe just add it as a parameter instead of adding it as a configuration? Not sure which way is the best. One concern might be that this is a static configuration that cannot be changed.

Reply:
(If we'd add it as a parameter, we will have to add it to `JavaRDDLike` too)